File: phase1/chapters/sorting_algorithms/topics/bubble_sort/sort.cpp
#include "iostream"

using namespace std;

void _swap(int *xp, int *yp)
{
    int temp = *xp;
    *xp = *yp;
    *yp = temp;
}

void _printArray(int arr[], int size)
{
    for (int i = 0; i < size; ++i)
        cout << arr[i] << " ";
    cout << endl;
}

void bubbleSort(int arr[], int arr_length)
{
    for (int i = 0; i < arr_length - 1; ++i)
    {
        for (int j = 0; j < arr_length - i - 1; ++j)
        {
            if (arr[j] > arr[j + 1])
            {
                _swap(&arr[j], &arr[j + 1]);
            }
        }
    }
}

int main()
{
    int arr[] = {64, 34, 25, 12, 22, 11, 90};
    int n = sizeof(arr) / sizeof(arr[0]);
    bubbleSort(arr, n);
    cout << "Sorted array: " << endl;
    _printArray(arr, n);
    return 0;
}

File: phase1/chapters/sorting_algorithms/topics/bubble_sort/implementation/sort.js
function bubbleSort(inputArr) {
  let arr_length = inputArr.length;
  for (let i = 0; i < arr_length - 1; ++i) {
    for (let j = 0; j < arr_length - i - 1; ++j) {
      if (inputArr[j] > inputArr[j + 1]) {
        let temp = inputArr[j];
        inputArr[j] = inputArr[j + 1];
        inputArr[j + 1] = temp;
      }
    }
    console.log(inputArr, inputArr[arr_length - i - 1]);
  }
  return inputArr;
}

function printArray(arr, size) {
  let i;
  for (i = 0; i < size; i++) console.log(arr[i]);
}

let arr = [64, 25, 12, 22, 11];
bubbleSort(arr);
console.log("Sorted array: \n");
printArray(arr);
